### PR TITLE
Studio supports allocation of agent server

### PR DESCRIPTION
**Pull request by pan-x-c**
### Auto server allocation
In the previous version, users must provide `host` and `port` for each `to_dist` call, if they want to use an existing agent server.
Now, the studio will automatically allocate a server to the `to_dist` function when the app is connected to the AgentScope Studio.

### Improve customized agent class loading
In the new version, the agent server also supports loading a directory of customized agent classes with in-directory dependencies with the `agent_dir` parameter.

### Update Tutorial

The tutorial for the above new features and the usage of server manager page has been updated.

## Checklist

Please check the following items before code is ready to be reviewed.

- [x]  Code has passed all tests
- [x]  Docstrings have been added/updated in Google Style
- [x]  Documentation has been updated
- [x]  Code is ready for review

### CODE REVIEW

File: docs/sphinx_doc/en/source/tutorial/208-distribute.md [hunk 1]
```diff
@@ -82,11 +82,15 @@ server.launch()
 server.wait_until_terminate()
 ```
 
-> For similarity, you can run the following command in your terminal rather than the above code:
->
-> ```shell
-> as_server --host ip_a --port 12001 --model-config-path model_config_path_a
-> ```
+For similarity, you can run the following command in your terminal rather than the above code:
```
Comment (qbc2016) [inline docs/sphinx_doc/en/source/tutorial/208-distribute.md:85]: For similarity？Do you mean simple?

File: docs/sphinx_doc/en/source/tutorial/208-distribute.md [hunk 2]
```diff
@@ -315,6 +322,65 @@ When running large-scale multi-agent applications, it's common to have multiple
...
+
+# Start the service
+server.launch()
+server.wait_until_terminate()
+```
+
+For scenarios using the command `as_server` in your command line, simply fill in the `--studio-url` parameter.
+
+```shell
+as_server --host ip_a --port 12001 --model-config-path model_config_path_a --agent-dir parent_dir_of_agent_a_and_b --studio-url http://studio_ip:studio_port
```
Comment (DavdGao) [inline docs/sphinx_doc/en/source/tutorial/208-distribute.md:354]: What's the relationship between the arguments `--agent-dir` and `custom_agent_classes`?

File: docs/sphinx_doc/en/source/tutorial/208-distribute.md [hunk 3]
```diff
@@ -315,6 +322,65 @@ When running large-scale multi-agent applications, it's common to have multiple
         ok = client.delete_all_agent()
     ```
 
+#### Connecting to AgentScope Studio
+
+The agent server process can be connected to [AgentScope Studio](#209-gui-en) at startup, allowing the `to_dist` method in subsequent distributed applications to be assigned automatically by Studio without the need for any parameters.
```
Comment (DavdGao) [inline docs/sphinx_doc/en/source/tutorial/208-distribute.md:327]: What if the server or the main process (where the user run the application) didn't connect to the studio ?

File: docs/sphinx_doc/zh_CN/source/tutorial/209-gui.md [hunk 4]
```diff
@@ -148,3 +148,59 @@ as_workflow config.json --compile ${YOUR_PYTHON_SCRIPT_NAME}.py
...
+</h1>
+
+#### Servers 列表
+
+注册到 Studio 的智能体服务器进程（Server）都会显示在 Server Manager 页面的 Servers 列表中，列表中会不仅会显示每个 Server 的 `ID`, `Hostname`, `Port`, `Created Time`，还会显示每个 Server 的状态以及计算资源使用情况，包括 `Status`, `CPU Usage`, `Memory Usage`。
+
+其中 `Status` 有以下几种：
+    - `running`：表示 Server 正在运行。
+    - `dead`：表示 Server 已停止运行。
+    - `unknown`：表示目前无法正常访问 Stduio 服务。
```
Comment (DavdGao) [inline docs/sphinx_doc/zh_CN/source/tutorial/209-gui.md:189]: ==> studio

File: src/agentscope/agents/agent.py [hunk 5]
```diff
@@ -64,7 +64,7 @@ def __call__(cls, *args: tuple, **kwargs: dict) -> Any:
                     ),
                     max_timeout_seconds=to_dist.pop(  # type: ignore[arg-type]
                         "max_timeout_seconds",
-                        1800,
+                        7200,
```
Comment (DavdGao) [inline src/agentscope/agents/agent.py:67]: Why we need to set the timout as 7200, which maybe too large

File: src/agentscope/agents/agent.py [hunk 6]
```diff
@@ -100,9 +100,9 @@ def __init__(
         host: str = "localhost",
         port: int = None,
         max_pool_size: int = 8192,
-        max_timeout_seconds: int = 1800,
+        max_timeout_seconds: int = 7200,
         local_mode: bool = True,
-        lazy_launch: bool = True,
+        lazy_launch: bool = False,
```
Comment (DavdGao) [inline src/agentscope/agents/agent.py:105]: Why we disable the lazy_launch here?

**Review by DavdGao — COMMENTED** [review]
See inline comments.

File: docs/sphinx_doc/en/source/tutorial/208-distribute.md (hunk 2 above)
Comment (pan-x-c) [reply docs/sphinx_doc/en/source/tutorial/208-distribute.md:354]: Almost the same, but we can not set `custom_agent_classes` in the command-line.

File: docs/sphinx_doc/en/source/tutorial/208-distribute.md (hunk 3 above)
Comment (pan-x-c) [reply docs/sphinx_doc/en/source/tutorial/208-distribute.md:327]: Degenerate to Subprocess mode

File: src/agentscope/agents/agent.py (hunk 5 above)
Comment (pan-x-c) [reply src/agentscope/agents/agent.py:67]: To ensure the user's right code can be run successfully without any additional parameters.

File: src/agentscope/agents/agent.py (hunk 6 above)
Comment (pan-x-c) [reply src/agentscope/agents/agent.py:105]: This feature is not used at all and may be removed in future versions.

File: src/agentscope/agents/rpc_agent.py [hunk 7]
```diff
@@ -69,8 +69,14 @@ def __init__(
         if agent_id is not None:
             self._agent_id = agent_id
         # if host and port are not provided, launch server locally
-        launch_server = port is None
+        if self.port is None and _studio_client.active:
+            server = _studio_client.alloc_server()
+            if "host" in server:
+                self.host = server["host"]
+                self.port = server["port"]
+        launch_server = self.port is None
```
[the remark this answers is not in the document]
Comment (pan-x-c) [reply src/agentscope/agents/rpc_agent.py:77]: The `port` will be checked by `check_port` of `utils.tools` before launching.

File: src/agentscope/server/launcher.py [hunk 8]
```diff
@@ -214,6 +227,68 @@ async def shutdown_signal_handler() -> None:
     )
 
 
+def load_agents_from_file(agent_file: str) -> list:
```
[the remark this answers is not in the document]
Comment (pan-x-c) [reply src/agentscope/server/launcher.py:230]: Sure, but to run customized agents, the code must be loaded.

**Review by xieyxclack — COMMENTED** [review]
LGTM

**Review by DavdGao — APPROVED** [review]

**Pull request merged**